nautobot/dcim/api/serializers.py

```diff
@@ -1160,17 +751,19 @@ def _get_termination(self, obj, side):
         if side.lower() not in ["a", "b"]:
             raise ValueError("Termination side must be either A or B.")
         termination = getattr(obj, f"termination_{side.lower()}")
-        serializer = get_serializer_for_model(termination, prefix="Nested")
-        context = {"request": self.context["request"]}
-        data = serializer(termination, context=context).data
-
-        return data
+        depth = int(self.context.get("depth", 0))
+        if depth == 0:
+            return termination.id
+        else:
+            relation_info = get_relation_info_for_nested_serializers(obj, termination, f"termination_{side.lower()}")
+            field_class, field_kwargs = self.build_nested_field(f"termination_{side.lower()}", relation_info, depth)
+            return field_class(termination, context={"request": self.context.get("request")}, **field_kwargs).data
```

Comment:
Is this a pattern that's common enough that we should create a utility function for it?